services/spar/test/Test/Spar/Sem/IdPSpec.hs

```diff
@@ -0,0 +1,158 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-# OPTIONS_GHC -fplugin=Polysemy.Plugin #-}
+
+module Test.Spar.Sem.IdPSpec where
```

Comment:
In order to run these on the cassandra interpreter, they would have to be defined under `.../src`, not `.../test`.  I propose `/services/spar/src/Spar/Sem/IdP/Spec.hs`.  It's not an interpreter, but I think ti's fine to put all interpreters and the properties that they share all in one directory.

Reply:
PTAL: please take a look :)

Reply:
I'll pull them out in a subsequent PR --- though we need the orphan instances from `Arbitrary ()`. OK to pull those too?